osu.Game/Overlays/Comments/CommentsContainer.cs

```diff
@@ -153,39 +156,44 @@ private void getComments()
             request?.Cancel();
             loadCancellation?.Cancel();
             request = new GetCommentsRequest(id.Value, type.Value, Sort.Value, currentPage++, 0);
-            request.Success += onSuccess;
+            request.Success += response => Schedule(() => OnSuccess(response));
             api.PerformAsync(request);
         }
 
-        private void clearComments()
+        protected void ClearComments()
         {
             currentPage = 1;
             deletedCommentsCounter.Count.Value = 0;
             moreButton.Show();
             moreButton.IsLoading = true;
             content.Clear();
+            CommentDictionary.Clear();
         }
 
-        private void onSuccess(CommentBundle response)
-        {
-            loadCancellation = new CancellationTokenSource();
+        protected readonly Dictionary<long, DrawableComment> CommentDictionary = new Dictionary<long, DrawableComment>();
 
-            LoadComponentAsync(new CommentsPage(response)
+        protected void OnSuccess(CommentBundle response)
+        {
+            if (!response.Comments.Any())
             {
-                ShowDeleted = { BindTarget = ShowDeleted },
-                Sort = { BindTarget = Sort },
-                Type = { BindTarget = type },
-                CommentableId = { BindTarget = id }
-            }, loaded =>
+                content.Add(new NoCommentsPlaceholder());
+                moreButton.Hide();
+                return;
+            }
+
+            var topLevelComments = AppendComments(response);
```

Comment:
Really not feeling this. Some comments are async-loaded while others parts aren't.

The loading part itself should be moved inside `AppendComments`. The other stuff done here can exist outside of the load.